rxandroidble/src/main/java/com/polidea/rxandroidble2/scan/ScanResult.java

```diff
@@ -1,10 +1,12 @@
 package com.polidea.rxandroidble2.scan;
 
+import androidx.annotation.NonNull;
 
 import com.polidea.rxandroidble2.RxBleDevice;
+import com.polidea.rxandroidble2.internal.ScanResultInterface;
 import com.polidea.rxandroidble2.internal.logger.LoggerUtil;
 
-public class ScanResult {
+public class ScanResult implements ScanResultInterface {
```

Comment:
One last thing I hope — could we have ~`ScanResultMock extends ScanResult implements ScanResultInterface`?
This could be a very trivial proxy class instantiated in `RxBleClientMock` and used in `*Mock` classes instead of `ScanResult`. This way `ScanResult`'s API would not change at all (no need to support it) and we would have all needed functionality.